frontend/model/contracts/group.js

```diff
@@ -918,8 +918,8 @@ sbp('chelonia/defineContract', {
           await sbp('gi.contracts/group/removeArchivedPayments', contractID)
 
           const groupIdToSwitch = Object.keys(contracts)
-            .find(cID => contracts[cID].type === 'gi.contracts/group' &&
-              cID !== contractID && rootState[cID].settings) || null
+            .filter(cID => contracts[cID].type === 'gi.contracts/group' && cID !== contractID)
+            .sort((cID1, cID2) => rootState[cID1].profiles?.[username] ? -1 : 1)[0] || null
```

Comment:
Same here, above line 920.

```js
// grab the groupID of any group that we've successfully finished joining
```


Also, you do not need to add `|| null` at the end, `groupIdToSwitch` will be set to `undefined` if it's not there.

Reply:
> Also, you do not need to add || null at the end, groupIdToSwitch will be set to undefined if it's not there.

Yeah, it'll be set to `undefined` if it's not there. But I can see that the type of `currentGroupId` is `string | null`. `null` value is used when there is no group as we can see in the `frontend/model/state.js`

Reply:
> null value is used when there is no group as we can see in the frontend/model/state.js

Hmm, OK, nice catch 👍